src/elements/content-explorer/ContentExplorer.js

```diff
@@ -411,6 +448,9 @@ class ContentExplorer extends Component<Props, State> {
         } else {
             this.setState({ rootName });
         }
+        if (this.shouldFetchThumbnails(items)) {
```

Comment:
It looks like this is repeated a few times. Can it live in `componentDidUpdate` and re-fetch based on when `currentCollection.items` changes?